kategory/src/main/kotlin/kategory/instances/IterableInstances.kt

```diff
@@ -0,0 +1,8 @@
+package kategory
+
+@Suppress("UNCHECKED_CAST")
+fun <A: Any, B> Iterable<A>.collect(vararg cases: PartialFunction<*, B>): List<B> {
```

Comment:
it's ok, handle Iterables and return a List? It's mandatory that  `collect()` return a  `List` or it could return something more generic like `Iterable` or `Collection`?